netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -326,6 +327,28 @@ public NettyChannelBuilder keepAliveWithoutCalls(boolean enable) {
     return this;
   }
 
+
+  /**
+   * If non-{@code null}, attempts to create connections bound to a local port.
+   */
+  public NettyChannelBuilder localSocketPicker(@Nullable LocalSocketPicker localSocketPicker) {
+    this.localSocketPicker = localSocketPicker;
+    return this;
+  }
+
+  @ExperimentalApi("FIXME")
+  public interface LocalSocketPicker {
```

Comment:
Probably better to be an abstract class to make it easier to change the interface in the future.

Reply:
Done.